src/components/MultiGestureCanvas/usePanGesture.js

```diff
@@ -0,0 +1,241 @@
+
+        const target = {
+            x: clamp(totalOffsetX.value, minVector.x, maxVector.x),
+            y: clamp(totalOffsetY.value, minVector.y, maxVector.y),
+        };
+
+        const isInBoundaryX = target.x === totalOffsetX.value;
+        const isInBoundaryY = target.y === totalOffsetY.value;
+
+        return {
+            target,
+            isInBoundaryX,
+            isInBoundaryY,
+            minVector,
+            maxVector,
+            canPanLeft: target.x < maxVector.x,
+            canPanRight: target.x > minVector.x,
+        };
+    }, [canvasSize.width, canvasSize.height]);
+
+    const returnToBoundaries = useWorkletCallback(() => {
+        const {target, isInBoundaryX, isInBoundaryY, minVector, maxVector} = getBounds();
+
+        if (zoomScale.value === zoomRange.min && totalOffsetX.value === 0 && totalOffsetY.value === 0 && panTranslateX.value === 0 && panTranslateY.value === 0) {
+            // we don't need to run any animations
+            return;
+        }
+
+        if (zoomScale.value <= zoomRange.min) {
+            // just center it
+            totalOffsetX.value = withSpring(0, SPRING_CONFIG);
+            totalOffsetY.value = withSpring(0, SPRING_CONFIG);
+            return;
+        }
+
+        if (isInBoundaryX) {
+            if (Math.abs(panVelocityX.value) > 0 && zoomScale.value <= zoomRange.max) {
+                totalOffsetX.value = withDecay({
+                    velocity: panVelocityX.value,
+                    clamp: [minVector.x, maxVector.x],
+                    deceleration: PAN_DECAY_DECELARATION,
+                    rubberBandEffect: false,
+                });
+            }
+        } else {
+            totalOffsetX.value = withSpring(target.x, SPRING_CONFIG);
+        }
+
+        if (!canPanVertically.value) {
+            totalOffsetY.value = withSpring(target.y, SPRING_CONFIG);
+        } else if (isInBoundaryY) {
+            if (
+                Math.abs(panVelocityY.value) > 0 &&
+                zoomScale.value <= zoomRange.max &&
+                // limit vertical pan only when content is smaller than screen
+                totalOffsetY.value !== minVector.y &&
+                totalOffsetY.value !== maxVector.y
+            ) {
+                totalOffsetY.value = withDecay({
+                    velocity: panVelocityY.value,
+                    clamp: [minVector.y, maxVector.y],
+                    deceleration: PAN_DECAY_DECELARATION,
+                });
+            }
+        } else {
+            totalOffsetY.value = withSpring(target.y, SPRING_CONFIG, () => {
+                isSwipingVertically.value = false;
+            });
+        }
+    });
+
+    const panGesture = Gesture.Pan()
+        .manualActivation(true)
+        .averageTouches(true)
+        .onTouchesMove((evt, state) => {
+            if (zoomScale.value > 1) {
+                state.activate();
+            }
+
+            // TODO: Swipe down to close carousel gesture
```

Comment:
What's the plan to get rid of this TODO? It seems there are nested TODOs. If the code isn't in use, I recommend removing the code and opening up a GH that explains the functionality that needs to be added.

Reply:
I extracted this pieces to https://github.com/Expensify/App/pull/33869